Merge my updated Client code into the main branch

ihalseide opened the pull request:
Updated the client to hopefully work with the server now

ihalseide committed:
make a nice user-interactive ship placement process and make the client get the server's outcome for a move

ihalseide committed:
remove server.txt variable

ihalseide committed:
remove 'display_board' function

ihalseide committed:
improve user info messages

ihalseide committed:
client aesthetic changes

Bleple commented:
currently there is a crash of the second client when it attempts to connect to the server. It looks to me like the client is trying to send a zero length message to the server, though how this is happening is unclear to me at the moment. The server doesn't crash, but does disconnect both clients due to the unhandled exception

ihalseide committed:
add extra checks to client

ihalseide committed:
rewrite the net message functions to make errors clearer

Bleple committed:
Update server.py

This should fix the issues with the server which became apparent by testing with this version of the client

ihalseide committed:
fix bug in sending messages that have multi-byte characters

ihalseide committed:
Merge pull request #13 from ihalseide/client-2_server-fixes

Update server.py for client-2

ihalseide commented:
Closed just in case the PR #12 wasn't included
